node/state/core.go

```diff
@@ -31,9 +33,25 @@ func CoreAccessor(endpoint string) func(fx.Lifecycle, keystore.Keystore, params.
 			return nil, err
 		}
 		signer := apptypes.NewKeyringSigner(ring, keyringAccName, string(net))
+		keys, err := signer.List()
+		if err != nil {
+			return nil, err
+		}
+		// if no key was found in keystore path, generate new key for node
+		if len(keys) == 0 {
```

Comment:
That's a working way to check if there are no keys, but is it possible to directly check if `keyringAccName` is present or no? Technically, there could be a case where the user copied some keys into the folder, and the keyring may see it here and does not go to the generation code path.

Reply:
?

Reply:
Why not just signer.Key(keyringAccName) != nil(simplified) ?